### PR TITLE
Added special casing for AWS Glacier

### PR DESCRIPTION
**Pull request by mattBrzezinski**
**NOTE: About 5 months ago I had made changes to the CI user to include GETs on Route53, I only realized this mistake now which is why that policy is bundled together here.**

## Problem
Currently any requests to AWS Glacier are failing. This is because it requires its API Version to be placed in the `x-amz-glacier-version` header, opposed to other services which want this value in `version`.

Other projects which have had this issue: [Swift AWS](https://github.com/swift-aws/aws-sdk-swift/issues/98), [Clojure AWS API](https://github.com/cognitect-labs/aws-api/issues/74)

Below is an example of the current state:
```julia
using AWSCore.Services
Services.glacier("GET", "/-/vaults")

HTTP.ExceptionRequest.StatusError(400, "GET", "/-/vaults", HTTP.Messages.Response:
"""
HTTP/1.1 400 Bad Request
x-amzn-RequestId: -J5ZjGzLNqL6lYuiyhjHMI0fYUWQ24tvFvZfUAqPUcQTlVI
Content-Length: 109
Date: Fri, 19 Jun 2020 17:44:07 GMT
Connection: close

{"code":"MissingParameterValueException","message":"Required parameter missing: API version","type":"Client"}""")
```

## Solution
I am modifying the `Services.glacier()` to add the `x-amz-glacier-version` header, instead of special casing in the `service_rest_json()`.

## Notes
Modifying the `Service.jl` file is improper as this file is autogenerated. However, the generation of this file is a manual process and as far as I am aware broken. Since I am working on archiving this package in favour of [AWS.jl](https://github.com/JuliaCloud/AWS.jl), I think directly modifying this file is an okay solution.

I spent some time, but could not come up with a better solution for adding in the glacier version header. There are two cases here to look at:

1) `args` do not contain a `headers` when passed in, and we want to add to this:

```julia
glacier("GET", "/-/vaults", ("limit"=>"1"))
```

2) `args` does contain `headers` when passed in, and we want to append to this:

```julia
glacier("GET", "/-/vaults", ("headers"=>["a"=>"b"]))
```

If anyone has a better solution, I'd love to hear!

### CODE REVIEW

**Comment by mattBrzezinski** [comment]
bors try

**Comment by mattBrzezinski** [comment]
Ugh... I'll update this users permissions when this is approved.

```
Create Vaults: Error During Test at /home/travis/build/JuliaCloud/AWSCore.jl/test/endpoints.jl:19
  Got exception outside of a @test
  403 -- User: arn:aws:iam::134847318362:user/AWSCoreJL is not authorized to perform: glacier:CreateVault on resource: arn:aws:glacier:us-east-1:134847318362:vaults/test-vault-01
```

**Comment by mattBrzezinski** [comment]
bors try

File: src/Services.jl [hunk 1]
```diff
@@ -1385,11 +1385,19 @@ gamelift(operation, args=[]) =
 gamelift(a...; b...) = gamelift(a..., b)
 
 function glacier(aws::AWSConfig, verb, resource, args=[])
+    version = "2012-06-01"
+    glacier_version_header = ["x-amz-glacier-version"=>version]
+
+    if args isa Pair && args.first == "headers"
+        append!(args[2], glacier_version_header)
+    else
+        args = vcat(args, ("headers" => glacier_version_header))
```
Comment (iamed2) [inline src/Services.jl:1394]: What if `args` is a dict? Or when it's a list-like collection, doesn't this drop any original version of "headers"?

File: src/Services.jl [hunk 2]
```diff
@@ -1385,11 +1385,19 @@ gamelift(operation, args=[]) =
 gamelift(a...; b...) = gamelift(a..., b)
 
 function glacier(aws::AWSConfig, verb, resource, args=[])
+    version = "2012-06-01"
+    glacier_version_header = ["x-amz-glacier-version"=>version]
+
+    if args isa Pair && args.first == "headers"
+        append!(args[2], glacier_version_header)
```
Comment (iamed2) [inline src/Services.jl:1392]: Should probably stick to either `args.first` and `args.second` or `args[1]` and `args[2]`.

File: src/Services.jl (hunk 1 above)
Comment (iamed2) [reply src/Services.jl:1394]: Tests using all sorts of ways of passing `args` would be helpful in verifying correct behaviour.

File: src/Services.jl [hunk 3]
```diff
@@ -1385,11 +1385,21 @@ gamelift(operation, args=[]) =
...
+    version = "2012-06-01"
+    glacier_version_header = Dict("headers"=>Dict("x-amz-glacier-version"=>version))
+
+    args = Dict(args)
+
+    if haskey(args, "headers")
+        args["headers"] = merge(+, args["headers"], glacier_version_header["headers"])
+    else
+        args = merge(args, glacier_version_header)
+    end
```
Comment (iamed2) [inline src/Services.jl:1397]:
```suggestion
    glacier_version_header = Dict("x-amz-glacier-version" => version)

    if haskey(args, "headers")
        args["headers"] = merge!(Dict(args["headers"]), glacier_version_header)
    else
        args["headers"] = glacier_version_header
    end
```
Headers doesn't have to be a dict either apparently. Also I don't think `+` does anything good there.

**Comment by mattBrzezinski** [comment]
Bors ran into a hiccup where it was getting `0` vaults on Linux 1.3. Re-running the job succeeded. I'm going to try this again.

**Pull request merged**